src/org/zaproxy/zap/extension/ascanrules/TestServerSideInclude.java

```diff
@@ -114,6 +116,10 @@ public void init() {
     public void scan(HttpMessage msg, String param, String value) {
         
 		if (this.inScope(Tech.Linux) || this.inScope(Tech.MacOS)) {
+			//Pre-check the original response for the detection pattern (to avoid false positives)
+			if (matchBodyPattern(getBaseMsg(), patternSSIUnix, new StringBuilder())) {

```

Comment:
- This should be done in the previous if statement (after tech checks), to not return earlier (Windows test could still be done). Also, worth extracting a method to reduce code duplication (`boolean isEvidencePresent(Pattern)`?).
 - Since the result is not required the `StringBuilder` can be `null`.